src/util/brands-url.ts

```diff
@@ -0,0 +1,29 @@
+const brandsBaseUrl = "https://brands.home-assistant.io/";
+const imageSuffix = ".png";
+
+export const brandsUrl = (
+  domain: string,
+  type: "icon" | "logo",
+  useFallback?: boolean
+) => {
+  let brandUrl = brandsBaseUrl;
+
+  if (useFallback) {
+    brandUrl += "_/";
+  }
+
+  brandUrl += domain + "/";
+
+  switch (type) {
+    case "icon":
+      brandUrl += "icon";
+      break;
+    case "logo":
+      brandUrl += "logo";
+      break;
+  }
+
+  brandUrl += imageSuffix;
+
+  return brandUrl;
+};
```

Comment:
This is overkill
```
return `https://brands.home-assistant.io/${useFallback ? '_/'  : ''}${domain}/${type}.png`
```

Reply:
And like the doc URL generator, this should have tests to ensure the output of the helper <https://github.com/home-assistant/frontend/blob/dev/test-mocha/util/generate-documentation-url.spec.ts>

Reply:
I'm a huge fan of splitting logic constants ("logo" | "icon" in this case) from other constants.
While they are identical atm, they may change in the future.

https://github.com/home-assistant/frontend/blob/4a22a7fd1ebc901af5192eb040302ef9e8053712/src/util/brands-url.ts#L17-L24
Having them seperated makes it easy to transparently modify them if it should ever be necessary

I do agree that this _could_ be shorter, however at least performance-wise this rather verbose approach shouldn't cause any issues because modern JS engines such as V8 will optimize that function so that it will end up like your suggestion.

I haven't run any benchmarks to confirm that of course

Reply:
We should not do string concatenation

Reply:
@bramkragten May I ask why?

While it's true that string concatenation is costly in some programming languages, this isn't the case for javascript

Reply:
Readability mostly

Reply:
I've added a benchmark: https://jsben.ch/ZiT6G

It's identical on FF 82. The only differences I'm seeing are caused by the OS scheduler and other software on this machine.
You may want to run it multiple times

Reply:
```js
  return `https://brands.home-assistant.io/${
    useFallback ? "_/" : ""
  }${domain}/${type}.png`;
```

This is what the prettier package.json script transforms the shorter expression to.

I personally wouldn't call this a readability improvement but here you go.